src/content/en/fundamentals/native-hardware/build-for-webusb/index.md

```diff
@@ -0,0 +1,969 @@
+  </tr>
+  <tr>
+   <td><code>0x0008</code></td>
+   <td>wLength</td>
+   <td>Size of this descriptor</td>
+  </tr>
+  <tr>
+   <td><code>0x0002</code></td>
+   <td>wDescriptorType</td>
+   <td>Function subset header descriptor</td>
+  </tr>
+  <tr>
+   <td><code>0x01</code></td>
+   <td>bFirstInterface</td>
+   <td>First interface of the function</td>
+  </tr>
+  <tr>
+   <td><code>0x00</code></td>
+   <td>bReserved</td>
+   <td>Must be set to 0</td>
+  </tr>
+  <tr>
+   <td><code>0x00A0</code></td>
+   <td>wSubsetLength</td>
+   <td>Total length of the subset including this header</td>
+  </tr>
+  <tr class="alt">
+   <td colspan="3">Microsoft OS 2.0 compatible ID descriptor</td>
+  </tr>
+  <tr>
+   <td><code>0x0014</code></td>
+   <td>wLength</td>
+   <td>Size of this descriptor</td>
+  </tr>
+  <tr>
+   <td><code>0x0003</code></td>
+   <td>wDescriptorType</td>
+   <td>Compatible ID descriptor</td>
+  </tr>
+  <tr>
+   <td><code>"WINUSB\0\0"</code></td>
+   <td>CompatibileID</td>
+   <td>ASCII string padded to 8 bytes</td>
+  </tr>
+  <tr>
+   <td><code>"\0\0\0\0\0\0\0\0"</codE></td>
+   <td>SubCompatibleID</td>
+   <td>ASCII string padded to 8 bytes</td>
+  </tr>
+  <tr class="alt">
+   <td colspan="3">Microsoft OS 2.0 registry property descriptor</td>
+  </tr>
+  <tr>
+   <td><code>0x0084</code></td>
+   <td>wLength</td>
+   <td>Size of this descriptor</td>
+  </tr>
+  <tr>
+   <td><code>0x0004</code></td>
+   <td>wDescriptorType</td>
+   <td>Registry property descriptor</td>
+  </tr>
+  <tr>
+   <td><code>0x0007</code></td>
+   <td>wPropertyDataType</td>
+   <td>REG_MULTI_SZ</td>
+  </tr>
+  <tr>
+   <td><code>0x002A</code></td>
+   <td>wPropertyNameLength</td>
+   <td>Length of the property name</td>
+  </tr>
+  <tr>
+   <td><code>"DeviceInterfaceGUIDs\0"</code></td>
+   <td>PropertyName</td>
+   <td>Property name with null terminator encoded in UTF-16LE</td>
+  </tr>
+  <tr>
+   <td><code>0x0050</code></td>
+   <td>wPropertyDataLength</td>
+   <td>Length of the property value</td>
+  </tr>
+  <tr>
+   <td><code>"{XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX}\0\0"</code></td>
+   <td>PropertyData</td>
+   <td>GUID plus two null terminators encoded in UTF-16LE</td>
+  </tr>
+</table>
+
+Windows will only query the device for this information once. If the device does
+not respond with valid descriptors it will not ask again the next time the
+device is connected. This
+[page](https://docs.microsoft.com/en-us/windows-hardware/drivers/usbcon/usb-device-specific-registry-settings){: .external}
+describes the registry entries created when enumerating a device. When testing
+it is useful to delete the entries created for a device in order to get Windows
+to try to read the descriptors again.
+
+For more information check out Microsoft's
+[blog post](https://blogs.msdn.microsoft.com/usbcoreblog/2012/09/26/how-to-install-winusb-sys-without-a-custom-inf/){: .external}
+on how to use these descriptors.
+
+## Feedback {: #feedback .hide-from-toc }
```

Comment:
You may want to include at the very end an example of this article such as https://github.com/sowbug/weblight?
WDYT?